substrate/client/db/src/lib.rs

```diff
@@ -1707,7 +1708,8 @@ impl<Block: BlockT> Backend<Block> {
 							&(start, end).encode(),
 						);
 					}
-				} else if number > best_num + One::one() &&
+				} else if operation.create_gap &&
```

Comment:
cc https://github.com/paritytech/polkadot-sdk/issues/4984

Reply:
Not exactly sure how it is related necessarily, it still had to pass through consensus successfully before ending up here IIUC.

I do see those "block has an unknown parent" in some cases though and subscribed to linked issue a while ago.